### PR TITLE
Test requirements

### PR DESCRIPTION
**Pull request by gerbyzation**
Addressing points 2, 3 and 4 of #3 

2. update test-core.ini with database, redis and solr settings.
3. CKAN is not located in a sibling directory relative to the manually installed extensions, the use setting needs to be updated in test.ini.
4. install dev/testing dependencies for specific plugin.

### CODE REVIEW

File: ckan/setup/start_ckan_development.sh [hunk 1]
```diff
@@ -19,20 +19,40 @@ do
...
     fi
 done
 
 # Update the plugins setting in the ini file with the values defined in the env var
 echo "Loading the following plugins: $CKAN__PLUGINS"
 paster --plugin=ckan config-tool $CKAN_INI "ckan.plugins = $CKAN__PLUGINS"
 
+# Update test-core.ini DB, SOLR & Redis settings
+echo "Loading test settings into test-core.ini"
+paster --plugin=ckan config-tool $SRC_DIR/ckan/test-core.ini "sqlalchemy.url = postgresql://ckan:ckan@db/ckan_test"
```
Comment (amercader) [inline ckan/setup/start_ckan_development.sh]: These databases should be created in the `db` image. And here we should probably use env vars for the connection strings to allow customization

**Comment by amercader** [comment]
@gerbyzation Thanks for this, see comment in #3 and changes in #5

File: ckan/setup/start_ckan_development.sh (hunk 1 above)
Comment (gerbyzation) [reply ckan/setup/start_ckan_development.sh]: Yes, will update this. I forgot to prune the volumes so these were still present in my DB.

**Comment by gerbyzation** [comment]
Have updated it to create test databases through initdb.d and read the urls from the environment.

File: ckan-dev/setup/start_ckan_development.sh [hunk 2]
```diff
@@ -37,6 +49,20 @@ python prerun.py
...
+paster --plugin=ckan config-tool \
+    $SRC_DIR/ckan/test-core.ini "ckan.datstore.write_url = $TEST_CKAN_DATASTORE_WRITE_URL"
+paster --plugin=ckan config-tool \
+    $SRC_DIR/ckan/test-core.ini "ckan.datstore.read_url = $TEST_CKAN_DATASTORE_READ_URL"
+
+paster --plugin=ckan config-tool $SRC_DIR/ckan/test-core.ini "solr_url = http://solr:8983/solr/ckan"
+paster --plugin=ckan config-tool $SRC_DIR/ckan/test-core.ini "ckan.redis_url = redis://redis:6379/1"
+
+# Run the prerun script to init CKAN and create the default admin user
+python prerun.py
```
Comment (amercader) [inline ckan-dev/setup/start_ckan_development.sh]: This is already called on line 46, no need to call it again right?

File: ckan-dev/setup/start_ckan_development.sh [hunk 3]
```diff
@@ -37,6 +49,20 @@ python prerun.py
...
+# Update test-core.ini DB, SOLR & Redis settings
+echo "Loading test settings into test-core.ini"
+paster --plugin=ckan config-tool \
+    $SRC_DIR/ckan/test-core.ini "sqlalchemy.url = $TEST_CKAN_SQLALCHEMY_URL"
+paster --plugin=ckan config-tool \
+    $SRC_DIR/ckan/test-core.ini "ckan.datstore.write_url = $TEST_CKAN_DATASTORE_WRITE_URL"
+paster --plugin=ckan config-tool \
+    $SRC_DIR/ckan/test-core.ini "ckan.datstore.read_url = $TEST_CKAN_DATASTORE_READ_URL"
+
+paster --plugin=ckan config-tool $SRC_DIR/ckan/test-core.ini "solr_url = http://solr:8983/solr/ckan"
```
Comment (amercader) [inline ckan-dev/setup/start_ckan_development.sh]: The Solr and Redis setting should also use env vars

File: ckan-dev/setup/start_ckan_development.sh [hunk 4]
```diff
@@ -37,6 +49,20 @@ python prerun.py
 echo "Loading the following plugins: $CKAN__PLUGINS"
 paster --plugin=ckan config-tool $CKAN_INI "ckan.plugins = $CKAN__PLUGINS"
 
+# Update test-core.ini DB, SOLR & Redis settings
+echo "Loading test settings into test-core.ini"
+paster --plugin=ckan config-tool \
```
Comment (amercader) [inline ckan-dev/setup/start_ckan_development.sh]: `config-tool` accepts passing multiple options to change at the same time, which will make things more readable:

```
paster --plugin=ckan config-tool $SRC_DIR/ckan/test-core.ini \
    "sqlalchemy.url = $TEST_CKAN_SQLALCHEMY_URL" \
    "ckan.datstore.write_url = $TEST_CKAN_DATASTORE_WRITE_URL" \
    "ckan.datstore.read_url = $TEST_CKAN_DATASTORE_READ_URL" \
    "solr_url = http://solr:8983/solr/ckan" \
    "ckan.redis_url = redis://redis:6379/1"
```

**Comment by amercader** [comment]
@gerbyzation can you also refactor against the changes in `.env.example`? cheers

File: ckan-dev/setup/start_ckan_development.sh (hunk 2 above)
Comment (gerbyzation) [reply ckan-dev/setup/start_ckan_development.sh]: Sorry, got pulled away and forgot to look it over before pushing it out

**Comment by amercader** [comment]
Thanks @gerbyzation!

**Pull request merged**